nameable.rb

```diff
@@ -0,0 +1,34 @@
+# Class Nameable
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+  end
+end
+
+# Class Decorator inhereting from Nameable
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+# Class CapitalizeDecorator inhereting from Decorator
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.split.map(&:capitalize).join(' ')
```

Comment:
- **[OPTIONAL]** This long method of chaining is somewhat unnecessary. Consider that we can implement this using the following code:

  ```
  @nameable.correct_name.capitalize
  ```